specification/resources/resource-manager/Microsoft.Authorization/2016-12-01/policyDefinitions.json

```diff
@@ -272,6 +421,13 @@
       "type": "string",
       "description": "The ID of the target subscription."
     },
+    "ManagementGroupIdParameter": {
+      "name": "managementGroupId",
+      "in": "path",
+      "required": true,
+      "type": "string",
+      "description": "The ID of the management group."
```

Comment:
Note: @sarangan12 is still the assigned reviewer. He will be finally reviewing the spec.
@vivsriaus - I noted an anomaly that I wanted to point out. Whenever you define a parameter in the global "prameters" section, Autorest treats it as a client property. This will end up as a required parameter in the constructor. I am sure this is not motive over here. It is ok to define parameters in the global parameters section so that you can reference it at multiple places in the swagger spec. However, if the parameter should not be a client property then please add the following extension: `"x-ms-parameter-location": "method"`.
There is an issue filed in the linter repo for checking this Azure/azure-openapi-validator#84

Azure/azure-openapi-validator#84

Reply:
Hey @amarzavery thanks for the comment. I'm not sure what a client property is. What do you mean when you say"required parameter in the constructor"? You mean of the policy client?

Reply:
Hey @vivsriaus - That is correct. The constructor of the policy client will have a required parameter named `managementGroupId`. An example TypeScript client:
```javascript
class PolicyClient {
 . . .
  constructor(subscriptionId: string, managementGroupId: string, baseUri: string?, options: OptionalParams) {
   . . .
  }
}
```
This would be the case in C# and other languages as well. I am sure this is not the desired behavior. If so then please add the extension `"x-ms-parameter-location": "method"` to the parameter in this and the other swagger spec below.

Reply:
Thanks @amarzavery. Yes, we definitely don't want managementGroupId to be a required parameter in the constructor. I'll make the changes.